### PR TITLE
docs: Add access_control_translation cfg block desc to aws_s3_bucket resource doc

### LINKED ISSUE #33894: [Docs]: s3_bucket access_control_translation
### Documentation Link

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_server_side_encryption_configuration

### Description

An argument of the destination configuration block, access_control_translation, isn't properly defined in the documentation. access_control_translation is recognized as a configuration block, and executing a terraform plan will throw an error if you define it as a regular argument.

Potential revision:
The `destination` configuration block supports the following arguments:
. . .

`access_control_translation` - (Optional) Specifies the overrides to use for object owners on replication . . . (documented below).
. . .

The `access_control_translation` configuration block supports the following arguments:

`owner` - (Required) Specifies the replica ownership . . .

### References

https://github.com/hashicorp/terraform/issues/34064
https://github.com/hashicorp/terraform/issues/23376

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by acwwat**
### Description

This PR adds more details for the `access_control_translation` configuration block that is relevant to replication configuration for the `aws_s3_bucket` resource.

### Relations

Closes #33894

### References

Referenced the [AccessControlTranslation API data type](https://docs.aws.amazon.com/AmazonS3/latest/API/API_AccessControlTranslation.html) for wordings.

### Output from Acceptance Testing

n/a

### CODE REVIEW

File: website/docs/r/s3_bucket.html.markdown [hunk 1]
```diff
@@ -220,11 +220,17 @@ The `destination` configuration block supports the following arguments:
 * `storage_class` - (Optional) The [storage class](https://docs.aws.amazon.com/AmazonS3/latest/API/API_Destination.html#AmazonS3-Type-Destination-StorageClass) used to store the object. By default, Amazon S3 uses the storage class of the source object to create the object replica.
 * `replica_kms_key_id` - (Optional) Destination KMS encryption key ARN for SSE-KMS replication. Must be used in conjunction with
   `sse_kms_encrypted_objects` source selection criteria.
-* `access_control_translation` - (Optional) Specifies the overrides to use for object owners on replication. Must be used in conjunction with `account_id` owner override configuration.
+* `access_control_translation` - (Optional) Specifies the overrides to use for object owners on replication ([documented below](#access-control-translation)). Must be used in conjunction with `account_id` owner override configuration.
```
Comment (justinretzolk) [inline website/docs/r/s3_bucket.html.markdown:223]: This, in conjunction with the other suggestion, will make sure that we don't wind up with duplicate anchor links

```suggestion
* `access_control_translation` - (Optional) Specifies the overrides to use for object owners on replication ([documented below](#access_control_translation-block)). Must be used in conjunction with `account_id` owner override configuration.
```

File: website/docs/r/s3_bucket.html.markdown [hunk 2]
```diff
@@ -220,11 +220,17 @@ The `destination` configuration block supports the following arguments:
...
 * `account_id` - (Optional) Account ID to use for overriding the object owner on replication. Must be used in conjunction with `access_control_translation` override configuration.
 * `replication_time` - (Optional) Enables S3 Replication Time Control (S3 RTC) ([documented below](#replication-time)).
 * `metrics` - (Optional) Enables replication metrics (required for S3 RTC) ([documented below](#metrics)).
 
+#### Access Control Translation
```
Comment (justinretzolk) [inline website/docs/r/s3_bucket.html.markdown:228]:
```suggestion
#### `access_control_translation` Block
```

**Review by justinretzolk — CHANGES_REQUESTED** [review]
A couple of minor suggestions to preserve anchor links, but otherwise looks good to me! Thank you for taking the time to contribute again!

**Comment by acwwat** [comment]
@justinretzolk Thanks for the review. I've made this one change, but there are 10s of other configuration blocks in the resource doc that are following the old style. Let me know if you'd like me to update the whole doc so that the style is consistent. Otherwise the PR is ready for your re-review.

**Review by justinretzolk — APPROVED** [review]
Hey @acwwat -- Let's merge the changes that you were targeting with this PR and then we can either open an issue so that we can come back for the others, or open a fresh PR to correct the rest. Since this is in good shape, I'll get it merged in shortly 🚀

**Pull request merged**